src/api/quotes.rs

```diff
@@ -0,0 +1,259 @@
+use crate::api::response::Response;
+use crate::Str;
+
+/// Filters quote data based on the timestamp in the given direction of time.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum QuoteTimespanFilter {
+    /// Filter for data older than the given timestamp.
+    LessThan,
+    /// Filter for data older than, or as old as, the given timestamp.
+    LessThanEqual,
+    /// Filter for data newer than the given timestamp.
+    GreaterThan,
+    /// Filter for data newer than, or as new as, the given timestamp.
+    GreaterThanEqual,
+}
+
+impl AsRef<str> for QuoteTimespanFilter {
+    fn as_ref(&self) -> &'static str {
+        match *self {
+            QuoteTimespanFilter::LessThan => "lt",
+            QuoteTimespanFilter::LessThanEqual => "lte",
+            QuoteTimespanFilter::GreaterThan => "gt",
+            QuoteTimespanFilter::GreaterThanEqual => "gte",
+        }
+    }
+}
+
+/// Specifies the sort order of the quotes, using the key specified in `sort`.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum QuoteOrder {
+    /// Order quotes oldest to newest.
+    Ascending,
+    /// Order quotes newest to oldests.
+    Descending,
+}
+
+impl AsRef<str> for QuoteOrder {
+    fn as_ref(&self) -> &'static str {
+        match *self {
+            QuoteOrder::Ascending => "asc",
+            QuoteOrder::Descending => "desc",
+        }
+    }
+}
+
+/// Specifies the sort key of the quotes, especially relevant for ordering (see `QuoteOrder`).
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum QuoteSortBy {
+    /// Sort quotes by timestamp.
+    Timestamp,
+}
+
+impl AsRef<str> for QuoteSortBy {
+    fn as_ref(&self) -> &'static str {
+        match *self {
+            QuoteSortBy::Timestamp => "timestamp",
+        }
+    }
+}
+
+/// A GET request to be made to the
+/// `/v3/quotes/<symbol>` endpoint.
+#[derive(Clone, Debug, PartialEq)]
+pub struct QuotesReq {
+    /// The ticker symbol to request quote data for.
+    pub symbol: String,
+    /// A date with the format YYYY-MM-DD or a nanosecond timestamp.
+    pub timestamp: String,
+    /// The optional timespan filter to use.
+    pub filter: Option<QuoteTimespanFilter>,
+    /// The optional ordering of the quotes.
+    pub order: Option<QuoteOrder>,
+    /// The optional sorting of the quotes.
+    pub sort: Option<QuoteSortBy>,
+}
+
+
+/// A ticker as returned by the
+/// `/v2/aggs/ticker/<symbol>/range/1/<span>/<start>/<end>` endpoint.
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct Quote {
+    /// The id of the exchange where this ask originates from.
+    pub ask_exchange: Num,
+    /// The ask price.
+    pub ask_price: Num,
+    /// The ask size.
+    pub ask_size: Num,
+    /// The id of the exchange where this bid originates from.
+    pub bid_exchange: Num,
+    /// The bid price.
+    pub bid_price: Num,
+    /// The bid size.
+    pub bid_size: Num,
+    #[serde(rename = "participant_timestamp")]
+    pub timestamp: Num,
```

Comment:
This timestamp should be decoded to `Datetime` like we do elsewhere. `Num` isn't really meant to be used for time stamps and provides not suitable methods for working with them.